nebula_decoders/include/nebula_decoders/nebula_decoders_velodyne/decoders/velodyne_generic_decoder.hpp

```diff
@@ -0,0 +1,445 @@
+  }
+
+  // DONE
+  void reset_overflow(double time_stamp)
+  {
+    if (overflow_pc_->points.size() == 0) {
+      scan_timestamp_ = -1;
+      overflow_pc_->points.reserve(max_pts_);
+      return;
+    }
+
+    // Compute the absolute time stamp of the last point of the overflow pointcloud
+    const double last_overflow_time_stamp =
+      scan_timestamp_ + 1e-9 * overflow_pc_->points.back().time_stamp;
+
+    // Detect cases where there is an unacceptable time difference between the last overflow point
+    // and the first point of the next packet. In that case, there was probably a packet drop so it
+    // is better to ignore the overflow pointcloud
+    if (time_stamp - last_overflow_time_stamp > 0.05) {
+      scan_timestamp_ = -1;
+      overflow_pc_->points.clear();
+      overflow_pc_->points.reserve(max_pts_);
+      return;
+    }
+
+    // Add the overflow buffer points
+    while (overflow_pc_->points.size() > 0) {
+      auto overflow_point = overflow_pc_->points.back();
+
+      // The overflow points had the stamps from the previous pointcloud. These need to be changed
+      // to be relative to the overflow's packet timestamp
+      double new_timestamp_seconds =
+        scan_timestamp_ + 1e-9 * overflow_point.time_stamp - last_block_timestamp_;
+      overflow_point.time_stamp =
+        static_cast<uint32_t>(new_timestamp_seconds < 0.0 ? 0.0 : 1e9 * new_timestamp_seconds);
+
+      scan_pc_->points.emplace_back(overflow_point);
+      overflow_pc_->points.pop_back();
+    }
+
+    // When there is overflow, the timestamp becomes the overflow packets' one
+    scan_timestamp_ = last_block_timestamp_;
+    overflow_pc_->points.clear();
+    overflow_pc_->points.reserve(max_pts_);
+  }
+
+  void unpack(const velodyne_msgs::msg::VelodynePacket & velodyne_packet)
+  {
+    const raw_packet_t * raw = (const raw_packet_t *)&velodyne_packet.data[0];
+    double last_azimuth_diff = 0;
+    uint16_t azimuth_next;
+    const uint8_t return_mode = velodyne_packet.data[RETURN_MODE_INDEX];
+    const bool dual_return = (return_mode == RETURN_MODE_DUAL);
+
+    int num_padding_blocks = sensor_.getNumPaddingBlocks(dual_return);
+
+    for (uint block = 0; block < static_cast<uint>(BLOCKS_PER_PACKET - num_padding_blocks);
+         block++) {
+      // Cache block for use.
+      const raw_block_t & current_block = raw->blocks[block];
+
+      uint bank_origin = 0;
+      // Used to detect which bank of 32 lasers is in this block.
+      switch (current_block.header) {
+        case BANK_1:
+          bank_origin = 0;
+          break;
+        case BANK_2:
+          bank_origin = 32;
+          break;
+        case BANK_3:
+          bank_origin = 64;
+          break;
+        case BANK_4:
+          bank_origin = 96;
+          break;
+        default:
+          // Do not flood the log with messages, only issue at most one
+          // of these warnings per minute.
+          return;  // bad packet: skip the rest
```

Comment:
Please do some error handling here, at least `RCLCPP_ERROR_THROTTLE()` or something to let the user know something is wrong.